### DISCUSSION TITLE
Forwarding rules cannot be set with a port OPNsense

### DISCUSSION THREAD

**Discussion by yolobnb** [DNSCrypt/dnscrypt-proxy#2556] [Q&A]
I have dnscrypt-proxy setup and running on my OPNsense router. However, I want to forward the queries to the plan IP on a non-default port (say :65353). I am not able to do this from the GUI in OPNsense. The following error comes up:

![image](https://github.com/DNSCrypt/dnscrypt-proxy/assets/98680372/e695cc69-0aa0-4fbc-aba8-784d09765f9c)

I am comfortable using the terminal to edit the configs and filters, but even if I edit it without GUI, the changes disappear soon after a reboot. I would appreciate any help.

**Comment by lifenjoiner** [comment]
I'm not OPNsense user. Learnt a bit from their official docs: their default [unbound](https://github.com/opnsense/docs/blob/master/source/manual/unbound.rst) adopts `@` to append a port ([unbound.conf(5)](https://nlnetlabs.nl/documentation/unbound/unbound.conf/)).

2 more guesses, if your screenshot is for `dnscrypt-proxy`, as I don't know it is for `unbound` or `dnscrypt-proxy`:
1. If it still works while "the changes disappear soon after a reboot", that is the issue of the GUI.
2. If it doesn't work any more or ever, you'd better try the official wizard from [OPNsense](https://github.com/opnsense/docs/blob/master/source/manual/how-tos/dnscrypt-proxy.rst) according the wiki. `dnscrypt-proxy` forwarding rules accept ports following `:`.

**Reply by yolobnb** [reply]
Thanks for the reply. The screenshot is the UI of `dnscrypt-proxy`. I don't use unbound and my plan was to use `dnscrypt-proxy` as my default DNS server and then forward the queries to another client. Unfortunately, the GUI doesn't accept ports as in the screenshot above. The manual edit of the `.txt` file is useless due to the reboot issue (it resets even when the service is restarted instead of a reboot, which is necessary for the changes to take effect). 

In the end, I did the reverse of what I wanted. The client first which upstream to `dnscrypt-proxy`. If the above request is still somehow possible, please provide an update.

**Reply by lifenjoiner** [reply]
> useless due to the reboot issue (it resets even when the service is restarted instead of a reboot, which is necessary for the changes to take effect)

`dnscrypt-proxy` does not modify nor reload the config files, as a back end.
It should be the issue of the (GUI) front end, which is not maintained here.
